x/rewards/keeper/iprpc.go

```diff
@@ -0,0 +1,170 @@
+package keeper
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
+	"github.com/lavanet/lava/x/rewards/types"
+)
+
+// handleNoIprpcRewardToProviders handles the situation in which there are no providers to send IPRPC rewards to
+// so the IPRPC rewards transfer to the next month
+func (k Keeper) handleNoIprpcRewardToProviders(ctx sdk.Context, iprpcReward types.IprpcReward) {
+	nextMonthIprpcReward, found := k.PopIprpcReward(ctx, false)
+	nextMonthId := k.GetIprpcRewardsCurrent(ctx)
+	if !found {
+		nextMonthIprpcReward = types.IprpcReward{Id: nextMonthId, SpecFunds: iprpcReward.SpecFunds}
+	} else {
+		nextMonthIprpcReward.SpecFunds = k.transferSpecFundsToNextMonth(iprpcReward.SpecFunds, nextMonthIprpcReward.SpecFunds)
+	}
+	k.SetIprpcReward(ctx, nextMonthIprpcReward)
+	details := map[string]string{
+		"transferred_funds":        iprpcReward.String(),
+		"next_month_updated_funds": nextMonthIprpcReward.String(),
+	}
+	utils.LogLavaEvent(ctx, k.Logger(ctx), types.TransferIprpcRewardToNextMonth, details,
+		"No provider serviced an IPRPC eligible subscription, transferring current month IPRPC funds to next month")
+}
+
+// countIprpcCu updates the specCuMap which keeps records of spec->SpecCuType (which holds the IPRPC CU per provider)
+func (k Keeper) countIprpcCu(specCuMap map[string]types.SpecCuType, iprpcCu uint64, spec string, provider string) {
+	if iprpcCu != 0 {
+		specCu, ok := specCuMap[spec]
+		if !ok {
+			specCuMap[spec] = types.SpecCuType{
+				ProvidersCu: map[string]uint64{provider: iprpcCu},
+				TotalCu:     iprpcCu,
+			}
+		} else {
+			_, ok := specCu.ProvidersCu[provider]
+			if !ok {
+				specCu.ProvidersCu[provider] = iprpcCu
+			} else {
+				specCu.ProvidersCu[provider] += iprpcCu
+			}
+			specCu.TotalCu += iprpcCu
+			specCuMap[spec] = specCu
+		}
+	}
+}
+
+// AddSpecFunds adds funds for a specific spec for <duration> of months.
+// This function is used by the fund-iprpc TX.
+func (k Keeper) addSpecFunds(ctx sdk.Context, spec string, fund sdk.Coins, duration uint64) {
+	startID := k.GetIprpcRewardsCurrent(ctx) + 1 // fund IPRPC only from the next month for <duration> months
+	for i := startID; i < startID+duration; i++ {
+		iprpcReward, found := k.GetIprpcReward(ctx, i)
+		if found {
+			// found IPRPC reward, find if spec exists
+			specIndex := -1
```

Comment:
instead of saving the specindex, just add the funds in the loop it self and set a boolean that you found it

Reply:
done. see https://github.com/lavanet/lava/commit/80e3986929c5d425d7fa24b3fb0cd674062a52cd